src/config.rs

```diff
@@ -45,7 +44,7 @@ pub struct CameraConfig {
     pub name: String,
 
     #[serde(rename = "address")]
-    pub camera_addr: SocketAddr,
+    pub camera_addr: String,
```

Comment:
This change removes all validation from parsing the camera address. (That is, if `SocketAddr` failed to parse, you'd get a "malformed config file" message, which I like.)  Question: Is there any validation that can be done, or is any string potentially a valid address?

Reply:
Yeah, so the problem here is that it's actually really hard to pass an IP/name that isn't valid. e.g. observe `ping 6`, `ping 6.6`. Even worse, the POSIX-specified name resolution interface (getaddrinfo) doesn't require any particular format, it only notes that DNS *might* be the provider.

I don't have any non-FreeBSD systems around to see if it holds for GNU/Linux, but a silly example that's perfectly valid and usable here is:
```
# ping 'x!foo'
ping: Unknown host
# echo '127.0.0.1 x!foo' >> /etc/hosts
# ping 'x!foo'
PING x!foo (127.0.0.1): 56 data bytes
64 bytes from 127.0.0.1: icmp_seq=0 ttl=64 time=0.203 ms
64 bytes from 127.0.0.1: icmp_seq=1 ttl=64 time=117.458 ms
...
```
I had considered doing something like the following, but had somewhat reached my daily limit of Rust at the time :-) e.g. (prepare for an abomination of a mix of pseudocode and an attempt at rust):

```
if addr matches "^[-A-za-z0-9.]:[0-9]+$" || addr.parse::<Ipv4Addr>().is_ok() || addr.parse::<Ipv6Addr>().is_ok() {
    ok
}
Err(...)
```

This avoids the problem I had in my earlier prototypes where my custom validator just .to_socket_addrs() that sucker and couldn't discern potentially transient DNS errors vs. validation errors.

IMHO it's worth validating; your userbase is probably incredibly unlikely to abuse the system resolver like that, and there's a lot more value in calling it an invalid hostname for those that wouldn't. Those of us that would fiddle with it will immediately recognize the issue if the verbiage is correct, and those that wouldn't fiddle will quickly see that there's a typo.

Reply:
This is a really good point. Had no idea you could put line noise in the hosts file (and have it work).

Regarding errors, I think either approach is okay as long as it loudly complains about unresolvable hostnames (in clear language - even "unresolvable hostname" is a bit obtuse). Since this is a runtime problem, and given your point above, I would lean toward accepting the config file and erroring on connect.

In other words, how you have it now :)

Reply:
I changed up the display string of the AddrResolutionError, but it might be too verbose: "Failed to translate camera address" -> I /think/ that strikes a decent balance between "what does that mean?" and "WHAT does that mean?" -- that might also be a step in the wrong direction, though. Thoughts?

Reply:
Yeah, it's tricky to come up with a good error message. I think that's a decent balance too. (I don't really have a better suggestion.) Looks good!